server/game/GameActions/ClaimFavorAction.ts

```diff
@@ -0,0 +1,36 @@
+import { PlayerAction, PlayerActionProperties} from './PlayerAction';
+import { EventNames } from '../Constants';
+import AbilityContext = require('../AbilityContext');
+import Player = require('../player');
+import { GameAction, GameActionProperties } from './GameAction';
+
+export interface ClaimFavorProperties extends PlayerActionProperties {
+    target: Player | null;
+}
+
+export class ClaimFavorAction extends PlayerAction {
+    name = 'discardFavor';
```

Comment:
Should this be `name = 'claimFavor'` ?  I'm not sure where the action names are used.